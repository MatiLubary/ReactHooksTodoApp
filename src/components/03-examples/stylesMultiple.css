body {
    background-color: navajowhite;
    padding: 15px;
}

.text-right{
    text-align: right;
}