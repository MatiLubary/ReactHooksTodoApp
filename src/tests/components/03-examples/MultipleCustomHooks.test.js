import { shallow } from 'enzyme'
import React from 'react'
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks'
import { useCounter } from '../../../hooks/useCounter'
import { useFetch } from '../../../hooks/useFetch'


jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')



describe('test in <MultipleCustomHooks />', () => {

    beforeEach( () => {
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        })
    })
    

    test('should display properly', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow( <MultipleCustomHooks /> );

        expect(wrapper).toMatchSnapshot();
        
    })


    test('should display the information', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'Mati',
                quote: 'hello world'
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow( <MultipleCustomHooks /> );

        expect(wrapper.find('.alert').exists()).toBe(false)
        expect(wrapper.find('p').text().trim()).toBe('hello world')
        expect(wrapper.find('footer').text().trim()).toBe('Mati')

        console.log(wrapper.html())
        
    })
    
    
})
