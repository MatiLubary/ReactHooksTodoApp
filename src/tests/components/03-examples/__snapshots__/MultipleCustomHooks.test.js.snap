// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <MultipleCustomHooks /> should display properly 1`] = `
<div>
  <h1>
    Breaking Bad quote #
    10
     
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;
